syntax = "proto3";

import "google/protobuf/empty.proto";
// import "google/protobuf/descriptor.proto";

package AiiiGRPC;

message Item {
    string name = 1;
    int32 price = 2;
    string message = 3;
}

message List {
    repeated Item items = 1;
}

message Filter {
    string name = 1;
    int32 price = 2;
    string opStr = 3;
}

message HealthCheckRes {
    int32 status_code = 1;
    string status_msg = 2;
}

service ToDoService {
    rpc HealthCheck (google.protobuf.Empty) returns (HealthCheckRes);
    rpc UnaryAddItem (Item) returns (List);
    rpc ClientStreamingAddItem (stream Item) returns (google.protobuf.Empty);
    rpc ServerStreamingSubList (Filter) returns (stream Item);
    rpc BidirectionalStreamingAsyncList (stream Item) returns (stream List);
}